wagtail/contrib/search_promotions/models.py

```diff
@@ -50,22 +76,54 @@ def garbage_collect(cls):
 
     @classmethod
     def get(cls, query_string):
+        """
+        # Gets or creates object matching query_string.
+
+        ## Args:
+            - query_string(str): Query text.
+
+        ## Returns:
+            - First Query object matching query_string.
+
+        ## To do:
+            - Rename get() to get_or_create()
+        """
         return cls.objects.get_or_create(
             query_string=normalise_query_string(query_string)
         )[0]
 
     @classmethod
     def get_most_popular(cls, date_since=None):
-        # TODO: Implement date_since
+        """
+        # Get most popular promotion, optionally since a date.
+
+        ## Args
+            - date_since (datetime.date, optional). Earliest date. Default is all.
+        """
+
+        objects = cls.objects.filter(daily_hits__isnull=False)
+
+        if date_since:
+            objects = objects.filter(daily_hits__date__gte=date_since)
+
```

Comment:
Thank you for this, however, it would be a bit easier to review if this was in a standalone PR. Would you be ok to pull this change out to a different PR?

Reply:
If possible, can you pull out these changes to a different PR? I think this change is best reviewed on its own.

I see you are using `main` for this PR so you may need to create a new branch off main locally and call it 'backup' or similar. That branch will contain all the work on `main`.

Then, switch back to the `main` branch and remove these changes.

Alternatively, if you want to keep pushing for this change in the same PR, it will need discrete unit tests added in wagtail/search/tests/test_queries.py for this work, along with maybe some data points to test against.

It's probably much simpler to stay focused on one feature per PR though, even though I know it's nice to try to fix some "TODO" while in the code.

**Update:** Looks like there are tests, thank you, maybe they can move to the file noted above that appears to be specifically for testing the queries.